### PR TITLE
New Accessibility Service depending on new native node module

### LINKED ISSUE #54349: Access keys not working as expected with custom titleBarStyle
- VSCode Version: 1.26.0 Insider (last commit date 2018-07-13)
- OS Version: Windows 10 April 2018 Update

**Steps to Reproduce**:

1. Ensure `Underline keyboard shortcuts and access keys` is checked in Windows' *Make the keyboard easier to use* options, under *Ease of Access* in the control panel.
2. In user settings, set the titleBarStyle to "custom" (`"window.titleBarStyle": "custom"`).
*Note: with the titleBarStyle set to `native`, keyboard navigation doesn't work as expected either.*

Does this issue occur when all extensions are disabled?: 
Yes

**Expected result**:
- Access keys are visible at all times, i.e. even before pressing the Alt key.

After pressing the Alt key:
- First menu item is focused and access keys remain visible when Alt key is released.
- When pressing an access key, the corresponding menu option is triggered/opened.
- When opening a menu item with the arrow keys and then pressing an access key, triggers/opens menu item.

**Actual result**:
- Access keys are not visible before pressing the Alt key.

After pressing the Alt key:
- First menu item is focused, but access keys immediately become invisible when Alt key is released.
- Pressing access key doesn't do anything.
- When opening a menu item with the arrow keys, access keys are visible again, but pressing them does nothing.

Only when holding the Alt key the entire time, keyboard navigations works as expected.

**Animated GIF's to compare behaviour with other Windows apps**:
See the difference between Visual Studio Code and Visual Studio (the latter works as expected, although you can basically use any classic Windows application, for example Notepad works the same way too).

Visual Studio Code:
vscodeaccesskeys
**Note**: I pressed down instead of the access key 'F' for *File* because the access key doesn't work.

Visual Studio:
vsaccesskeys

### PR DESCRIPTION
**Pull request by sbatten**
replaces https://github.com/Microsoft/vscode/pull/65529
fixes https://github.com/Microsoft/vscode/issues/54349

### CODE REVIEW

**Review by bpasero — CHANGES_REQUESTED** [review]
As discussed, I suggest to rather introduce a `IAccessibilityService` and not use "throw Error" for conditional programming. I would also throw in the functionality of [`getAccessibilitySupport`](https://github.com/Microsoft/vscode/blob/a02649bb7b9b440cd4b8a57fba4780079c4ff1e2/src/vs/base/browser/browser.ts#L132) and [`onDidChangeAccessibilitySupport`](https://github.com/Microsoft/vscode/blob/a02649bb7b9b440cd4b8a57fba4780079c4ff1e2/src/vs/base/browser/browser.ts#L135)

File: package.json [hunk 1]
```diff
@@ -52,6 +52,7 @@
     "vscode-ripgrep": "^1.2.5",
     "vscode-sqlite3": "4.0.7",
     "vscode-textmate": "^4.0.1",
+    "vscode-windows-registry": "^1.0.0",
```
Comment (bpasero) [inline package.json]: @sbatten should this move into `optionalDependencies`?

**Review by bpasero — APPROVED** [review]
@sbatten I like this. I will add @alexandrudima as reviewer given the changes for the standalone editor.

**Comment by bpasero** [comment]
Maybe one thing to test is the performance impact of accessing the windows registry on startup. @alexandrudima are we already doing this on Windows for keybindings? Do we have an idea about the timing?

**Comment by bpasero** [comment]
@sbatten would be great if you could also try to use registerSingleton and move the service out of workbench + have it as lazy service.

**Review by alexdima — APPROVED** [review]
:+1: Cool! Thanks for remembering to define a service for the standalone editor

**Comment by alexandrudima** [comment]
@bpasero Yeah, we are going into `node-native-keymap` on startup and I think that ends up calling that. AFAIK the Windows registry is very very fast, but perhaps the native call overhead is noticeable, I haven't checked...

**Pull request merged**